File: NOVEMBER 23/22 November/Symmetric Tree gfg.cpp
class Solution{
    public:
    // return true/false denoting whether the tree is Symmetric or not
    bool check(struct Node* left, struct Node* right)
    {
        if(!left && !right)
            return true;
        if(!left || !right || left->data != right->data)
            return false;
        return check(left->right, right->left) && check(left->left, right->right);
    }
    bool isSymmetric(struct Node* root)
    {
	    // Code here
	    if(!root)
	        return true;
	    return check(root->left, root->right);
    }
};

File: JUNE 2025/14 June/Symmetric Tree gfg.java
class Solution {
    boolean checkSymmetry(Node left, Node right)
    {
        if(left==null && right==null)
            return true;
        
        if((left==null || right==null) || left.data!=right.data)
            return false;
        
        return checkSymmetry(left.right, right.left) && checkSymmetry(left.left, right.right);
    }
    
    public boolean isSymmetric(Node root) {
        // Code here
        return checkSymmetry(root.left, root.right);
    }
}
